components/devicecloud-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/GroupManagerService.java

```diff
@@ -0,0 +1,252 @@
+        Group group = new Group();
+        group.setId(groupId);
+        group.setName(name);
+        group.setDescription(description);
+        group.setOwnerId(username);
+        group.setDateOfLastUpdate(new Date().getTime());
+        try {
+            new GroupManagement().getGroupManagementService().updateGroup(group);
+            return true;
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}")
+    @DELETE
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean deleteGroup(@PathParam("groupId") int groupId, @FormParam("username") String username) {
+        try {
+            return new GroupManagement().getGroupManagementService().deleteGroup(groupId);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Group getGroup(@PathParam("groupId") int groupId, @FormParam("username") String username) {
+        try {
+            return new GroupManagement().getGroupManagementService().getGroupById(groupId);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/name/{groupName}")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Group getGroupByName(@PathParam("groupName") String groupName, @FormParam("username") String username) {
+        try {
+            return new GroupManagement().getGroupManagementService().getGroupByName(groupName);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/all")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Group[] getGroupsOfUser(@FormParam("username") String username) {
+        try {
+            List<Group> groups = new GroupManagement().getGroupManagementService().getGroupListOfUser(username);
+            Group[] groupArray = new Group[groups.size()];
+            return groups.toArray(groupArray);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/all/count")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public int getGroupCountOfUser(@FormParam("username") String username) {
+        try {
+            return new GroupManagement().getGroupManagementService().getGroupCountOfUser(username);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
```

Comment:
Remove printStackTraces, use proper log messages and return proper status codes here.